thunder/core/interpreter.py

```diff
@@ -605,6 +606,13 @@ def push_frame_stack(self, frame: InterpreterFrame):
             pf = self._pop_frame_stack()
             assert pf is frame, "Frame stack inconsistency"
 
+    def get_current_user_source_location(self) -> tuple[str, Positions]:
```

Comment:
If I use `thunder.torch.softmax` why is the source location recorded as
```py
thunder/torch/__init__.py:4411: 	    return _softmax(a, dim=dim, dtype=dtype)
```
and with `torch.softmax` it points to the function being traced:
```py
<ipython-input-14-72fa81ca31be>:2: 	    return torch.softmax(a, 0)
```

Reply:
This is because the "symbol boundary" for softmax is `_softmax` apparently (the symbols are opaque functions to the JIT components, so they are the topmost frame). Note that softmax is special in how we define it in `thunder.torch` (to cater for the undocumented parameter). 
I see three options (for a follow-up PR):
- refuse to record locations in thunder.torch (deferring to the previous (next lower) stack frame if we are),
- provide a mechanism to fudge this (mark things "internal", might also be worthwile for "impl" functions if we see them),
- do nothing.

Reply:
Filed https://github.com/Lightning-AI/lightning-thunder/issues/475